API/API/Controllers/ProductController.cs

```diff
@@ -1,109 +1,96 @@
-﻿using Microsoft.EntityFrameworkCore;
+﻿using API.General.Classes.Configure;
+using API.Models;
+using API.Services.Interfaces;
 using Microsoft.AspNetCore.Mvc;
-using API.Context;
-using API.Services.Clases;
 
 // <summary>
 // Developer....: Karla Ramos Benitez       USER ID: XKRB
 // </summary>
-namespace API.Controllers
+namespace API.Controllers;
+
+/// <summary>
+/// Products controllers
+/// </summary>
+[Route("/products")]
+[ApiController]
+public class ProductController : ControllerBase
 {
+
+    private readonly IProductService? _productService;
+
+    public ProductController(IProductService productService) => _productService = productService;
+
     /// <summary>
-    /// Products controllers
+    /// Get product data
     /// </summary>
-    [Route("/Products")]
-    [ApiController]
-    public class ProductController:ControllerBase
+    /// <param name="GetProduct"> product Id </param>
+    /// <returns> product Id  </returns>
+    [HttpGet("{GetProduct}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    public async Task<ActionResult<ProductModel>> GetProduct(int GetProduct)
     {
-        private readonly ProductContext _context;
-
-        public ProductController(ProductContext context)
+        try
         {
-            _context = context;
+            ProductModel product = await _productService.GetProduct(GetProduct);
+            return Ok(product);
         }
-
-        /// <summary>
-        /// Get product data
-        /// </summary>
-        [HttpGet]
-        public async Task<ActionResult<ProductItem>> GetProductItem(int id)
+        catch (ExceptionClass ex)
         {
-            var productItem = await _context.ProductItems.FindAsync(id);
-
-            if (productItem == null)
-            {
-                return NotFound();
-            }
-            return productItem;
+            return BadRequest(ex.Message);
         }
+    }
 
-        /// <summary>
-        /// Post product data
-        /// </summary>
-        [HttpPost]
-        public async Task<ActionResult<ProductItem>> PostProductItem(ProductItem productItem)
+    /// <summary>
```

Comment:
The comments is not complete, does not have the parameter and return section

Reply:
Done